content/compute-app/non-functional-requirements/readme.md

```diff
@@ -0,0 +1,156 @@
+  - Example functional requirement: A customer can submit an invoice for payment
+  - Example non-functional requirement: The system is available and accepts invoices for payment during 99.9% of business hours
+
+- **Types of requirements.** Not all requirements (especially NFRs) are the same.
+  - Some requirements are specified contractually or in regulation, compliance standards, or legislation; a violation of those requirements might cause financial or legal problems.
+   - Other requirements are implied, but very important - for example, an ISV might not explicitly commit to a certain level of uptime, but if their solution is offline for three hours every day then their customers will lose patience and go across the street.
+   - Some other requirements are hidden - for example, if you're an ISV, you might not know until it's too late that one of your customers assumes they will have their own database, and when they discover that you share the database with other      tenants it causes a major problem.
+   - And some requirements are made up by the business, and don't really have any direct consequence. For example, you might aim to have your APIs respond within 500ms. If you miss the mark, nobody is going to sue you, but the business cares about this because it's part of their quality and performance goal.
+   - Distinguish SLAs (service level agreements) vs. SLOs (service level objectives).
+  - For each requirement, it's helpful to have a clear understanding of whether the requirement is an SLA or an SLO, who "owns" the requirement, who is responsible for monitoring compliance with the requirement, who is responsible for rectifying any violations, and what happens if it's violated. A lot of the time these aren't written down or understood systematically, but they should be.
+
+- **Tradeoffs**. An important part of solution architecture is deciding on tradeoffs. For example, imagine you're building an enterprise solution. Your security team wants you to lock down your environment, but your development team is telling you that will cause them major problems and make everything take much longer, which means your timeframes will be at risk. What should you do?
+
+- **WAF pillars.** Talk about the fact that the five WAF pillars are there to help to structure our thinking and conversations about the key areas of NFRs.
+
+  The rest of this session focuses on each of the pillars in turn, and talks about the key NFRs in each, using them both as a way to explain the requirements and also as a way to think systematically about non-functional requirements overall.
+
+## Reliability
+
+- **Pillar Concept** - HA/BC/DR - scenarios, availability and recoverability
```

Comment:
When discussing reliability, customers sometimes conflate and confuse HA and DR. I think it maybe good to provide a good distinction between these concepts and which metrics apply to each (e.g., SLA in HA, RPO/RTO in DR).

Reply:
Absolutely - we will ensure we clearly distinguish availability from disaster recovery.